src/Sarif.Multitool.Library/OptionsInterpretter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    public class OptionsInterpretter
+    {
+        //  Poor man's dependency injection
+        public OptionsInterpretter() : this (new EnvironmentVariableGetter())
+        {
+
+        }
+        
+        public OptionsInterpretter(IEnvironmentVariableGetter environmentVariableGetter)
+        {
+            if (environmentVariableGetter == null)
+            {
+                throw new ArgumentException("Required parameter", nameof(environmentVariableGetter));
+            }
+
+            _environmentVariableGetter = environmentVariableGetter;
+        }
+
+        private readonly IEnvironmentVariableGetter _environmentVariableGetter;
+
+        public void ConsumeEnvVarsAndInterpretOptions(AnalyzeOptionsBase analyzeOptionsBase)
+        {
+            //  Level and Kind are the only two properties we're reading from env variables for now.  See if they are populated.
+            IEnumerable<ResultKind> kindsFromEnv = getOptionEnumFromEnvVar<ResultKind>(nameof(analyzeOptionsBase.Kind));
+            IEnumerable<FailureLevel> levelsFromEnv = getOptionEnumFromEnvVar<FailureLevel>(nameof(analyzeOptionsBase.Level));
+
+            //  Union in C# is distinct
+            analyzeOptionsBase.Kind = analyzeOptionsBase.Kind.Union(kindsFromEnv);
```

Comment:
>Union(kindsFromEnv); [](start = 62, length = 20)

do we need to check for null?

Reply:
Done.

---
In reply to: [566073724](https://github.com/microsoft/sarif-sdk/pull/2273#discussion_r566073724) [](ancestors = 566073724)